### ISSUE TITLE
Replace `@dual-bundle/import-meta-resolve` with `import-meta-resolve`

### ISSUE THREAD

**Issue by ybiquitous** [stylelint/stylelint#7556]
In PR #7532, we introduced the [`@dual-bundle/import-meta-resolve`](https://www.npmjs.com/package/@dual-bundle/import-meta-resolve) package to support both ESM and CJS, which is a fork of [`import-meta-resolve`](https://www.npmjs.com/package/import-meta-resolve) that is pure ESM.

When we switch to pure ESM (maybe in the next major version), we can replace `@dual-bundle/import-meta-resolve` with `import-meta-resolve`.

See also the code below:

https://github.com/stylelint/stylelint/blob/a211d8bb2bd2d4c5319852ef4fb35c83f1a82bc3/lib/utils/resolveSilent.mjs#L6-L7

> [!NOTE]
> Node.js has provided the [`import.meta.resolve()`](https://nodejs.org/api/esm.html#importmetaresolvespecifier) API, and we would like to use the API instead of `import-meta-resolve`, but it is still experimental. In addition, the `--experimental-import-meta-resolve` flag is necessary to use the second argument `parent` (at the point of Node.js 21.7.1). So it may be difficult to switch `import.meta.resolve()` in the next major version.

**Comment by boris-petrov** [comment]
Upgrading to Node.js 22.0.0 and running `Stylelint` leads to `(node:202332) [DEP0180] DeprecationWarning: fs.Stats constructor is deprecated.`. It comes from `@dual-bundle/import-meta-resolve/lib/resolve.js:139:12`. So replacing this package is important to also fix this (if the other one has already resolved that deprecation).

**Comment by JounQin** [comment]
@boris-petrov I believe you're describing https://github.com/wooorm/import-meta-resolve/issues/27, right?

**Comment by boris-petrov** [comment]
@JounQin yes. It's good there is an issue so hopefully it will be fixed at some point. :)

**Comment by ybiquitous** [comment]
https://github.com/wooorm/import-meta-resolve/releases/tag/4.1.0 now resolved the deprecation warning.

@JounQin Could you merge the upstream change into `@dual-bundle/import-meta-resolve`, please?

**Comment by JounQin** [comment]
`@dual-bundle/import-meta-resolve@4.1.0` has been released.

cc @boris-petrov

**Comment by ybiquitous** [comment]
In Node.js 20.6.0, `import.meta.resolve()` has been supported without the experimental flag:
<img width="853" height="60" alt="Image" src="https://github.com/user-attachments/assets/6cc3619f-96a1-4091-97d8-dcb31c5b724f" />

So, we don't need both npm packages.

Ref:
- https://nodejs.org/api/esm.html#importmetaresolvespecifier
- https://github.com/stylelint/stylelint/blob/a8c356eaad7d40770e167ff3aef753da775d7f3d/lib/utils/resolveSilent.mjs#L6-L7
- https://github.com/stylelint/stylelint/blob/a8c356eaad7d40770e167ff3aef753da775d7f3d/lib/utils/resolveSilent.mjs#L59-L65
- https://github.com/stylelint/stylelint/pull/8867
- https://github.com/stylelint/stylelint/pull/7532

**Comment by ybiquitous** [comment]
Oh, I misunderstood the `--experimental-import-meta-resolve` flag. The doc says: 

> When using the `--experimental-import-meta-resolve` flag, that function accepts a second argument:

So, if we continue using the second argument `parent`, the experimental flag is required. 😓

**Comment by jeddy3** [comment]
Closed by https://github.com/stylelint/stylelint/pull/8886

**Comment by romainmenke** [comment]
See: https://github.com/nodejs/node/issues/61127
We might eventually get a native node API
